lib/_http_client.js

```diff
@@ -197,7 +197,9 @@ function ClientRequest(options, cb) {
     self = null;
   });
 
-  this._ended = false;
+  self._ended = false;
+  self.res = null;
```

Comment:
Perhaps we should also change `delete req.res` later in this file to `req.res = null`?